extensions/app-runtime.md

```diff
@@ -0,0 +1,51 @@
+# Application Runtime Extensions
+
+This document specifies a contract between a platform and processes running in an application OCI image.  
+
+## Runtime Environment Variables
+
+The following environment variables extend the [Environment - Provided by the Platform](../buildpack.md#provided-by-the-platform) section in the Buildpack Interface Specification.
+The are exclusively available during the launch phase.
+
+| Env Variable         | Example Value                    | Description                       | Mandatory
+|----------------------|----------------------------------|----------------------------------------------
+| `PORT`               | `8080`                           | port for primary web process type | [x]
+| `CNB_APP_ROUTES`     | `{"web":{"http":{"port":8080}}}` | routes for each process type      | [x]
+| `CNB_APP_INDEX`      | `0`                              | app instance index (logical)      | [x] 
+| `CNB_APP_NAME`       | `example-app`                    | app name                          |
```

Comment:
I think that `CNB_APP_NAME` should be required.  Each platform has the idea of a logical identifier for a running processes and I think we should require that exposure.  The key problem with leaving it optional is that for one of it’s strongest use-cases (self-identification in a remote system) I don’t think there’s a good default that buildpacks could use if it was missing.

Reply:
No strong opinion about this. I'll change it to mandatory.

Reply:
https://github.com/buildpack/spec/pull/42/commits/0f76577dc60ab6d06cfb9ca08600988288602d11